pkg/router/template/template_helper.go

```diff
@@ -62,6 +62,14 @@ func isInteger(s string) bool {
 	return (err == nil)
 }
 
+func toInt(s string) int {
+	ret, err := strconv.Atoi(s)
+	if err != nil {
+		return 0
```

Comment:
Since this is becoming a part of the informal API for route templates, we should consider how it might be used or expected to work in the general case.  Is returning 0 for invalid input a sensible default? Or should we return `nil`? For example, we could do the following:
```go
func toInt(s string) *int {
	ret, err := strconv.Atoi(s)
	if err != nil {
		return nil
	}
	return &ret
}
```
Which could be used as follows:
```
{{- with $threads := toInt (env "ROUTER_THREADS") }}
  ￼{{- if gt $threads 0 }}
￼    nbthread {{ $threads }}
  {{- end }}
￼{{-  end }}
```
The part inside the `with` block will get skipped entirely if `strconv.Atoi` returns an error.

Reply:
Maybe document it as private (FWIW)?

I think this is another example of where a new field on an object passed to the template helps. There's no reason to do conversion inside the template itself: why would a user ever want to treat ROUTER_THREADS as anything but an int? If I were making a custom template and had to deal with this conversion manually I'd be annoyed